### PR TITLE
Ensures the managed loader is loaded only once.

### PR DESCRIPTION
**Pull request by tonyredondo**
This PR adds an Interlocked.CompareExchange to the loader IL code to ensure the loader is loaded only once.

@DataDog/apm-dotnet

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 1]
```diff
@@ -1467,6 +1467,135 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id,
...
+  pALNewInstr = rewriter_already_loaded.NewILInstr();
+  pALNewInstr->m_opcode = CEE_LDC_I4_1;
+  rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+  // ldc.i4.0 : Load the constant 0 (int) to the stack
+  pALNewInstr = rewriter_already_loaded.NewILInstr();
+  pALNewInstr->m_opcode = CEE_LDC_I4_0;
+  rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+  // call System.AppDomain System.AppDomain.CurrentDomain property
```
Comment (kevingosse) [inline src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:1569]: I think you mean Interlocked.CompareExchange :D

**Review by kevingosse — CHANGES_REQUESTED** [review]
Would it be possible to add an integration test for this? (have an application that reproduces the situation in which the loader is called multiple times, and check that the loading occurs only once)

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp (hunk 1 above)
Comment (tonyredondo) [reply src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:1569]: ups, you're right

Comment (tonyredondo) [reply src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp]: fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1055/commits/2c168c30f743bbfcbdc44b8c75d27e37195053c2

**Comment by tonyredondo** [comment]
> Would it be possible to add an integration test for this? (have an application that reproduces the situation in which the loader is called multiple times, and check that the loading occurs only once)

done in https://github.com/DataDog/dd-trace-dotnet/pull/1055/commits/c2b6b09c66ff5e37d73db45d9a4111ccd5972214

**Comment by tonyredondo** [comment]
The result of the integration test logs running in master:
```ini
[Samples.NoMultiLoader.exe] 12156: [info] Datadog CLR Profiler 1.20.0 32-bit
[Samples.NoMultiLoader.exe] 12156: [info] Environment variables:
[Samples.NoMultiLoader.exe] 12156: [info]   DD_TRACE_ENABLED=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_TRACE_DEBUG=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_DUMP_ILREWRITE_ENABLED=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_DOTNET_TRACER_HOME=C:\Users\danielredondo\.dotnet\tools\.store\dd-trace\1.20.0\dd-trace\1.20.0\tools\netcoreapp3.1\any\..\..\..\home
[Samples.NoMultiLoader.exe] 12156: [info]   DD_INTEGRATIONS=C:\Users\danielredondo\.dotnet\tools\.store\dd-trace\1.20.0\dd-trace\1.20.0\tools\netcoreapp3.1\any\..\..\..\home\integrations.json
[Samples.NoMultiLoader.exe] 12156: [info]   DD_PROFILER_PROCESSES=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_PROFILER_EXCLUDE_PROCESSES=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_AGENT_HOST=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_TRACE_AGENT_PORT=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_ENV=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_SERVICE=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_VERSION=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_DISABLED_INTEGRATIONS=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_CLR_DISABLE_OPTIMIZATIONS=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_TRACE_DOMAIN_NEUTRAL_INSTRUMENTATION=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_TRACE_NETSTANDARD_ENABLED=
[Samples.NoMultiLoader.exe] 12156: [info]   DD_AZURE_APP_SERVICES=
[Samples.NoMultiLoader.exe] 12156: [info]   APP_POOL_ID=
[Samples.NoMultiLoader.exe] 12156: [info]   DOTNET_CLI_TELEMETRY_PROFILE=
[Samples.NoMultiLoader.exe] 12156: [info] Profiler attached.
[Samples.NoMultiLoader.exe] 12156: [info] JITCompilationStarted: Startup hook registered in function_id=140702017220848 token=100663297 name=Samples.NoMultiLoader.Deps.DatabaseSample.CreateSqlConnection(), assembly_name=Samples.NoMultiLoader.Deps app_domain_id=1755093778480 domain_neutral=0
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] AssemblyLoadFinished: Datadog.Trace.ClrProfiler.Managed v1.20.0 matched profiler version v1.20.0
[Samples.NoMultiLoader.exe] 12156: [info] *** JITCompilationStarted() replaced calls from System.Net.Http.HttpMessageInvoker.SendAsync() to System.Net.Http.HttpMessageHandler.SendAsync() 100664409 with calls to Datadog.Trace.ClrProfiler.Integrations.HttpMessageHandlerIntegration.HttpMessageHandler_SendAsync() 167773897
[Samples.NoMultiLoader.exe] 12156: [info] *** JITCompilationStarted() replaced calls from System.Net.Http.HttpClientHandler.SendAsync() to System.Net.Http.HttpMessageHandler.SendAsync() 100664409 with calls to Datadog.Trace.ClrProfiler.Integrations.HttpMessageHandlerIntegration.HttpMessageHandler_SendAsync() 167773897
[Samples.NoMultiLoader.exe] 12156: [info] *** JITCompilationStarted() replaced calls from System.Net.Http.HttpClientHandler.SendAsync() to System.Net.Http.HttpMessageHandler.SendAsync() 100664409 with calls to Datadog.Trace.ClrProfiler.Integrations.HttpMessageHandlerIntegration.HttpMessageHandler_SendAsync() 167773897
[Samples.NoMultiLoader.exe] 12156: [info] *** JITCompilationStarted() replaced calls from System.Net.Http.SocketsHttpHandler.SendAsync() to System.Net.Http.HttpMessageHandler.SendAsync() 100664409 with calls to Datadog.Trace.ClrProfiler.Integrations.HttpMessageHandlerIntegration.HttpMessageHandler_SendAsync() 167773897
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
[Samples.NoMultiLoader.exe] 12156: [info] ModuleLoadFinished: Datadog.Trace.ClrProfiler.Managed.Loader loaded into AppDomain 1755093778480 clrhost
...
```

**Review by kevingosse — APPROVED** [review]
LGTM

File: test/test-applications/integrations/dependency-libs/Samples.NoMultiLoader.Deps/Samples.NoMultiLoader.Deps.csproj [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <ApiVersion Condition="'$(ApiVersion)' == ''">4.7.0</ApiVersion>
+  </PropertyGroup>
+
+
+  <ItemGroup Condition="$(TargetFramework.StartsWith('net4'))">
+    <Reference Include="System.Data" />
+  </ItemGroup>
+
+  <ItemGroup Condition="!$(TargetFramework.StartsWith('net4'))">
+    <PackageReference Include="System.Data.SqlClient" Version="$(ApiVersion)" />
```
Comment (zacharycmontoya) [inline test/test-applications/integrations/dependency-libs/Samples.NoMultiLoader.Deps/Samples.NoMultiLoader.Deps.csproj:19]: Super small nit: If you know that this version of the NuGet package triggers the issue and you don't plan to test other versions, feel free to remove the `ApiVersion` property

**Review by zacharycmontoya — APPROVED** [review]
Looks good!

File: test/test-applications/integrations/dependency-libs/Samples.NoMultiLoader.Deps/Samples.NoMultiLoader.Deps.csproj (hunk 2 above)
Comment (tonyredondo) [reply test/test-applications/integrations/dependency-libs/Samples.NoMultiLoader.Deps/Samples.NoMultiLoader.Deps.csproj]: done in https://github.com/DataDog/dd-trace-dotnet/pull/1055/commits/2bc41ac1d3bdba5a1eb989335800a9d59e2f9e3c

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 3]
```diff
@@ -1467,6 +1467,135 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id,
     return hr;
   }
 
+  /////////////////////////////////////////////
+  // Define a new static field _isAssemblyLoaded on the new type type that has a int value.
```
Comment (lucaspimentel) [inline src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:1471]:
```suggestion
  // Define a new static int field _isAssemblyLoaded on the new type.
```

**Review by lucaspimentel — APPROVED** [review]
LGTM, just a nit in a code comment

File: test/test-applications/integrations/dependency-libs/Samples.NoMultiLoader.Deps/Samples.NoMultiLoader.Deps.csproj [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">net45;net461;netstandard2.0;netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netstandard2.0;netcoreapp3.1</TargetFrameworks>
```
Comment (lucaspimentel) [inline test/test-applications/integrations/dependency-libs/Samples.NoMultiLoader.Deps/Samples.NoMultiLoader.Deps.csproj:5]: I don't think we need to target all these framework in a dependency lib, but that's out of scope of this PR.

**Pull request merged**